### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Salwa99**
# Blog app - integration specs for views and fix n+1 problems

### Hello there!

**In this PR I:**

- Handle the  n+1 issues.

**## here are screenshots of the logs:**

**Before fixing the N+1 problem**
before

**After fixing the N+1 problem**
after

- Add tests for `users` / `posts` views.

- Fix Linter errors.

- Follow GitFlow.

_Cheers! :heart:_

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class PostsController < ApplicationController
   before_action :fetch_author
 
   def index
-    @posts = @user.posts
+    @posts = Post.all
   end
```
Comment (DammyShittu) [inline app/controllers/posts_controller.rb]: - [ ] In demonstrating the elimination of N+1 query problem, you need to consider that we want to fetch all posts, but we also what to fetch their respective comments along with them. Kindly modify your index action here to do that. See example.

  ```
  @posts = @user.posts.includes(:comments)
  ```

File: spec/features/user_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  describe 'Check each user' do
+    it "Redirect to user1's show page" do
+      click_link("user-#{user1.id}")
+      expect(current_path).to eq user_path(user1)
+    end
+    it "Redirect to user2's show page" do
+      click_link user2.name
+      expect(current_path).to eq user_path(user2)
+    end
+  end
```
Comment (DammyShittu) [inline spec/features/user_index_spec.rb]: - [ ] For the user index page, you are yet to add test for this scenario

  image

File: spec/features/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      expect(current_path).to eq user_post_path(user, post3)
+    end
+  end
+
+  describe 'Click to see all posts' do
+    it "redirects to user's post's index page" do
+      click_link 'See All Posts'
+      expect(current_path).to eq user_posts_path(user)
+    end
+  end
```
Comment (DammyShittu) [inline spec/features/user_show_spec.rb]: - [ ] For the user show page, kindly test for this case.

  image

File: spec/features/post_index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    end
+
+    it 'should have a button for pagination' do
+      expect(page).to have_button('Pagination')
+    end
+  end
+
+  describe 'Clicking on a post' do
+    it "should redirect to correct post's show page" do
+      click_link post2.title
```
Comment (DammyShittu) [inline spec/features/post_index_spec.rb]: - [ ] Kindly add tests for asserting we can see the user's profile picture.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Salwa99 and @bill7pearl,

## Highlights
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the required changes by the previous reviewer

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**